Workflow

JohanLossius opened the pull request:
unit tests and cypress tests added, along with various changes.

resit with deadline the 21st of April, after failing the first attempt for the workflow CA/module.

Commit:
test file only

Commit:
prettier installation

Commit:
prettier config

Commit:
1.0.1

Commit:
install eslint

Commit:
config eslint

Commit:
script lint

Commit:
eslint work

Commit:
1.0.2

Commit:
commit automation eslint

Commit:
lint automation fix, scss directory not sure correct

Commit:
1.0.3

Commit:
test v2 commit lint stage

Commit:
lint stage setup and test.js fix

Commit:
commit jest and eslint config, and remove test.js file

Commit:
test jest eslint staging

Commit:
test prettier on index.js

Commit:
babel for jest config

Commit:
cypress halfway config

Commit:
install all dependencies, babel, jest, cypress, eslint, etc. Cypress not working still.

Commit:
1.1.0

Commit:
install newest sass version

Commit:
1.1.1

Commit:
fix index html test title. commit after aborting revert of pre sass commit

Commit:
readme file added

Commit:
recent jest and cypress work

Commit:
cypress works after dependency and packages installations. libnss, libasound, etc.

Commit:
1.2.0

Commit:
eslintrc.json commit

Commit:
commit after cypress config and readme.md workflow status bdage files, except not cypress folder

Commit:
commit after try fix eslint for cypress

Commit:
change title index.html

Commit:
add unit test files

Commit:
readme modifications

Commit:
cypress eslintrc.json file

Commit:
add cypress files and cypress.config.js and .estlinrc.json fixes

Commit:
1.2.1

Commit:
update for package.json file

Commit:
readme updates

Commit:
readme file

Commit:
updated cypress tests after feedback

JohanLossius committed:
Delete index.test.js

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update readme.md

JohanLossius committed:
Update and rename readme.md to README.md

JohanLossius committed:
Update cypress.config.js

Connor got error message when running the cypress app, so fixed that, by adding supportFile: false,

Fermain commented on .eslintrc.json:
```diff
@@ -0,0 +1,35 @@
+    "extends": "eslint:recommended",
+    "overrides": [
+      {
+        "files": ["**/*.test.js", "**/*.cy.js"],
```
This is not quite correct. You should have two objects in this array, one for cypress and one for jest. Here you have merged them into a single object.

JohanLossius replied:
Ok, I'll sort that out.
I guess it better clarity with 2 separate objects.
Thanks!

Fermain commented on index.js:
```diff
@@ -1,5 +1,5 @@
-ui()
-router()
\ No newline at end of file
+ui();
```
Formatting changes clutter this PR, there are 100 file changes but I only care about 11-15 of them

JohanLossius replied:
Aha, that's fine. Thanks for feedback.

Fermain reviewed (changes requested):
Only one minor change is required

JohanLossius committed:
Update .eslintrc.json

make file overrides for jest and cypress into 2 different objects after feedback from oliver
